### PR TITLE
acl fee sheet fix

### PR DESCRIPTION
**Pull request by bradymiller**

### CODE REVIEW

**Comment by bradymiller** [comment]
@matrix-amiel ,
Here's proposed fix for the acl issue on the fee sheet.

> The admin superbill acl is for the administration->codes gui and should not be used here. This form is already controlled by the encounters->coding aco and that is what should be used. Also not a fan of this formJump since it's silent; much better to show a message that not authorized.

File: interface/forms/fee_sheet/new.php [hunk 1]
```diff
@@ -29,10 +29,11 @@
 require_once("codes.php");
 
 //acl check
-$can_view = acl_check('admin', 'superbill', false, 'view');
-$can_write = acl_check('admin', 'superbill', false, 'write');
-if (!$can_view && !$can_write)
+if (!acl_check('encounters','coding'))
```
Comment (bradymiller) [inline interface/forms/fee_sheet/new.php]: Will instead use the acl_check_form() function here. This is something we could then add to all the forms to further secure them.

**Comment by bradymiller** [comment]
Testing very well so will bring into codebase.

**Comment by shachar058** [comment]
@bradymiller 
In order to improve our fixes on acl and give us a better understanding - is there some documentation on the permissions and what they do?
Because right now if we try to do even the slightest fix in ACL we're kinda working in the dark...
Thanks!

**Comment by bradymiller** [comment]
@shachar058 ,

Yep, this stuff is not very intuitive :)

I started a doc on ACL awhile back (needs to be updated with Rod and yours recent ACL work and newer mechanisms(such as being able to assign an ACO to forms, document categories, and CDR rules)):
http://www.open-emr.org/wiki/index.php/Access_Controls_Listing

I usually figure it out by searching via the code and trying to keep it consistent when use a current ACO (or when adding an ACO). We should keep the following sheet updated though and also is very important to keep track of the return values that are used for each ACO (write, addonly, etc.).

-brady

**Comment by shachar058** [comment]
@bradymiller 

Ok, thanks.

I see the doc speaks about 'Return Value' too. What do you mean by the 'return value'?
I also see that some permissions have thngs like 'wsome,addonly optional'  in paranthesis.
Is this connected? What does it all mean?

Thanks for the guidance.

**Comment by bradymiller** [comment]
@shachar058 ,

Quick overview of ACL:
ACO - These are the items that are being secured. For example, the admin->acl ACO is needed for a user to be able to use Administration->ACL gui.
ACL - This is a list of ACOs. There are 2 concepts here that can cause confusion. The first is that the ACL is set to return a value if it contains the ACO (in OpenEMR, use write, view, addonly, wsome). The second is that the ACL is assigned to a group(ie. Administrators, Accounting).
Group - This is what you actually assign users to.

So, when you do an acl_check() (https://github.com/openemr/openemr/blob/master/library/acl.inc#L163), you are checking if the user has access to an ACO. If you don't assign to check for specific return value(s) in acl_check (ie. 4th parameter), then you pass with any return value (note that much of the codebase actually does this).

And regarding the wiki doc for the acl's. Definitely the more specific and detailed we can be, the better.

**Comment by shachar058** [comment]
@bradymiller 
Thank you!

**Pull request merged**